### PR TITLE
Service implementation

### PR DESCRIPTION
**Pull request by napalmpapalam**

### CODE REVIEW

File: internal/services/api/main.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+			handlers.CtxStorage(cfg.Storage()),
+		),
+	)
+
+	r.Route("/v1", func(r chi.Router) {
+		r.Route("/proofs", func(r chi.Router) {
+			r.Use(handlers.AuthMiddleware())
+			r.Post("/", handlers.ProofCreate)
+			r.Route("/{id}", func(r chi.Router) {
+				r.Get("/", handlers.ProofByID)
```
Comment (artemskriabin) [inline internal/services/api/main.go:34]: get must be public (without auth)

File: internal/cli/main.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	case migrateDownCmd.FullCommand():
+		if err := MigrateDown(cfg); err != nil {
+			panic(errors.Wrap(err, "failed to migrate down"))
+		}
+	default:
+		panic(fmt.Errorf("unknown command %s", cmd))
+	}
+
+	gracefulStop := make(chan os.Signal, 1)
+	signal.Notify(gracefulStop, syscall.SIGTERM, syscall.SIGINT)
```
Comment (artemskriabin) [inline internal/cli/main.go:83]: you can you notify context
https://gitlab.com/q-dev/indexer-svc/-/blob/master/cmd/indexer/main.go?ref_type=heads#L35

Comment (artemskriabin) [reply internal/cli/main.go:83]: and wait for context.Done()

File: internal/services/api/main.go (hunk 1 above)
Comment (napalmpapalam) [reply internal/services/api/main.go:34]: fixed

File: internal/cli/main.go (hunk 2 above)
Comment (napalmpapalam) [reply internal/cli/main.go:83]: fixed

**Review by artemskriabin — APPROVED** [review]

**Pull request merged**